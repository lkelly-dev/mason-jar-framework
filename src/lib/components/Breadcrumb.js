import React from 'react';
import { cx } from 'emotion';
import MODIFIERS from './Modifiers';
import { debugAttrs, debugMode } from '../utils/debugging';

const Breadcrumb = props => {
  const { context, children, className, ...rest } = props;
  const debug = debugMode(context)
  const breadcrumbClass = cx('breadcrumb', className);
  return (
    <div className={breadcrumbClass}
      {...(debug && debugAttrs(props)) }
      {...rest}>
      {children}
    </div>
  );
};

const BreadcrumbItem = props => {
  const { context, children, status, className, ...rest } = props;
  const debug = debugMode(context)
  const breadcrumbStatus = MODIFIERS.BREADCRUMB[status] ? `${status}` : null;
  const breadcrumbClass = cx('breadcrumb-item', breadcrumbStatus, className);
  return (
    <div className={breadcrumbClass}
      {...(debug && debugAttrs(props)) }
      {...rest}>
      {children}
    </div>
  );
};

Breadcrumb.Item = BreadcrumbItem;

export { Breadcrumb, BreadcrumbItem };
