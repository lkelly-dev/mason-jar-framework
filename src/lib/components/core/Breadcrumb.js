import React from 'react';
import { cx } from 'emotion';
import MODIFIERS from './Modifiers';

const Breadcrumb = props => {
  const { children, className, ...rest } = props;
  const breadcrumbClass = cx('breadcrumb', className);
  return (
    <div className={breadcrumbClass} {...rest}>
      {children}
    </div>
  );
};

const BreadcrumbItem = props => {
  const { children, status, className, ...rest } = props;
  const breadcrumbStatus = MODIFIERS.BREADCRUMB[status] ? `${status}` : null;
  const breadcrumbClass = cx('breadcrumb-item', breadcrumbStatus, className);
  return (
    <div className={breadcrumbClass} {...rest}>
      {children}
    </div>
  );
};

Breadcrumb.Item = BreadcrumbItem;

export { Breadcrumb, BreadcrumbItem };
