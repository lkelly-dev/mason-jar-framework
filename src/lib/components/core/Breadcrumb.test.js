import React from 'react';
import { shallow } from 'enzyme';
import { Breadcrumb, BreadcrumbItem } from "./Breadcrumb";

describe('Breadcrumb', () => {
  it('Breadcrumb matches snapshot', () => {
    const component = shallow(<Breadcrumb></Breadcrumb>);
    expect(component).toMatchSnapshot();
  })
});
