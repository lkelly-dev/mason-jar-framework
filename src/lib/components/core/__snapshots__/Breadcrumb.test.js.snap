// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb Breadcrumb matches snapshot 1`] = `
<div
  className="breadcrumb"
/>
`;
